src/engine/enginebuffer.cpp

```diff
@@ -1120,7 +1121,26 @@ void EngineBuffer::processSlip(int iBufferSize) {
 
     // Increment slip position even if it was just toggled -- this ensures the position is correct.
     if (enabled) {
-        m_dSlipPosition += static_cast<double>(iBufferSize) * m_dSlipRate;
+        double slipDelta = static_cast<double>(iBufferSize) * m_dSlipRate;
+        // a regular loop is active
+        if (m_pLoopingControl->isLoopingEnabled() &&
+                !m_pLoopingControl->isLoopRollActive()) {
+            double newPos = m_dSlipPosition + slipDelta;
+            // simulate looping
+            const double loopStart = m_pLoopingControl->getLoopSamples().start;
+            const double loopEnd = m_pLoopingControl->getLoopSamples().end;
+            while (newPos > loopEnd) {
```

Comment:
```suggestion
            while (newPos >= loopEnd) {
```
LoopEnd is the trigger point, the first point after the loop. 
Not sure if we need also consider the playback direction here, probably not because the while loop works in both directions.

Reply:
This would move us to loop_in.
`newPos <= loopStart` below would move us back to loop_end